src/lib/hooks/resource-names.svelte.ts

```diff
@@ -25,15 +25,25 @@ interface Context {
   current: Record<PartID, Query | undefined>;
 }
 
+const revisions = new Map<string, string>();
+
+const deepEqualResourceNames = (
+  a: ResourceName[],
+  b: ResourceName[]
+): boolean => {
+  if (a.length !== b.length) {
+    return false;
+  }
+
+  return a.every((item, i) => JSON.stringify(item) === JSON.stringify(b[i]));
```

Comment:
Should we sort object fields like we do when we render diffs to ensure we are making equal comparisons?

Reply:
Yes! good catch